docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+
+A validating webhook for the `AdoptedResource` custom resources will validate the existence of the target type in the cluster and ensure the requested type is able to be adopted. 
+
+The target metadata field is an arbitrary mapping of key-value pairs that will be applied as the metadata for the created custom resource. If a `name` field is specified in this mapping, the manager should use this value as the custom resource `name` . Otherwise, the manager will use the name specified in the metadata of the `AdoptedResource` as the default name for the resource. This same method will also apply to the `namespace` field. All other fields in the metadata mapping are applied to the created resource without validation.
+
+### Example Specifications
+
+**1. An adopted S3 bucket with the minimal number of fields**
+
+```
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: AdoptedResource
+metadata:
+  name: adopted-data-bucket
+spec:
+   apiVersion: s3.services.k8s.aws/v1alpha1
+   kind: Bucket
+   identifier:
+        arn: arn:aws:s3:::my-data-bucket
+```
+
+**2. An adopted S3 bucket with overriding metadata fields**
+
+```
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: AdoptedResource
+metadata:
+  name: adopted-data-bucket
+spec:
+   apiVersion: s3.services.k8s.aws/v1alpha1
+   kind: Bucket
+   metadata:
+        name: data-bucket
+        namespace: team-a
+        annotations:
+            ...
+   identifier:
+        arn: arn:aws:s3:::my-data-bucket
+```
+
+**3. An adopted API Gateway v2 API (using an ID identifier)**
+
+```
+apiVersion: apigateway.services.k8s.aws/v1alpha1
+kind: AdoptedResource
+metadata:
+  name: adopted-api-gateway
+spec:
+   apiVersion: apigateway.services.k8s.aws/v1alpha1
+   kind: Api
+   identifier:
+        id: 123456789012
+```
+
+### Custom Resource Validation Webhook
+
+In order to minimise the chance of user’s facing unexpected behaviours, a validating webhook will validate any adopted resource changes prior to submission to the ACK managers. The webhook will primarily reject the following error cases:
+
+1. The provided target ACK type does not exist within the cluster
+2. The target resource would override an existing resource
+3. The target resource type cannot be adopted
+
+## Adopted K8s Resource Tagging
+
+All adopted resources will be annotated with an ACK-specific annotation that differentiates it from a natively created resource. This annotation serves to ensure clarity for users about possible specification differences introduced during the adoption process, or for filtering by adopted status. However, once adopted into a cluster, a resource will be reconciled with the same logic as any natively created resource. 
+
+The proposed k8s adopted tag is `services.k8s.aws/adopted: true`
+
+The adopted annotation would remain on the custom resource for the remainder of its lifecycle, and should not be modified by any other manager operation. 
```

Comment:
What's the desire driving this limitation?

On the one hand, I get it: when I am in process of migrating from say Terraform to the ACK I want to have the labels and easily identify the resources. Makes total sense.

On the other hand: it's 1 year later, I am no longer using Terraform or care about the adoption process. I still have the `adopted: true` tag which is now redundant. Is there a use for a tag after the adoption process is complete?

---

Thinking more on this, "_not be modified by any other manager operation_" just blocks the ACK Manager from changing it, not the user. So the user could delete the tag? I am a bit confused.

Reply:
The idea behind this was as your first case points out, I don't want anything to wipe away the fact that this was an adopted resource in case we need that information for debugging or logical grouping.

However, I think it's totally fine if the user wants to delete it manually. I just wouldn't want some controller deciding when is appropriate.